contracts/SpokePool.sol

```diff
@@ -636,10 +634,39 @@ abstract contract SpokePool is
         uint32 fillDeadline,
         bytes memory message
     ) public payable override nonReentrant unpausedDeposits {
-        // Validate
+        // Check that deposit route is enabled for the input token. There are no checks required for the output token
+        // which is pulled from the relayer at fill time and passed through this contract atomically to the recipient.
+        require(enabledDepositRoutes[inputToken.token][destinationChainId], "Disabled route");
 
-        // Do stuff
-        // - Pull funds from depositor
+        //@dev The following checks are copied from _deposit. Are they still needed?
+        require(inputToken.amount <= MAX_TRANSFER_SIZE, "Amount too large");
+        require(outputToken.amount <= MAX_TRANSFER_SIZE, "Amount too large");
+
+        // Require that quoteTimestamp has a maximum age so that depositors pay an LP fee based on recent HubPool usage.
+        // It is assumed that cross-chain timestamps are normally loosely in-sync, but clock drift can occur. If the
+        // SpokePool time stalls or lags significantly, it is still possible to make deposits by setting quoteTimestamp
+        // within the configured buffer. The owner should pause deposits if this is undesirable. This will underflow if
+        // quoteTimestamp is more than depositQuoteTimeBuffer; this is safe but will throw an unintuitive error.
+
+        // slither-disable-next-line timestamp
+        require(getCurrentTime() - quoteTimestamp <= depositQuoteTimeBuffer, "invalid quoteTimestamp");
+
+        // fillDeadline is relative to the destination chain.
+        // Don’t allow fillDeadline to be more than ~3 bundles into the future, so 12 hours maybe.
+        // This way the dataworker/relayer doesn’t have to maintain more a lookback longer than this.
+        // TODO: We could choose to skip this check in favor of adding a rule in the dataworker to enforce this
+        // constraint.
+        require(fillDeadline <= getCurrentTime() + 12 hours, "invalid fillDeadline");
```

Comment:
nit: prefer making this an immutable variable that can be set on deployment, so it can, theoretically, be changed on redeploy.